snuba/admin/clickhouse/querylog.py

```diff
@@ -0,0 +1,54 @@
+from snuba.admin.clickhouse.common import (
+    InvalidCustomQuery,
+    get_ro_query_node_connection,
+    validate_ro_query,
+)
+from snuba.admin.notifications.querylog.audit_log import QuerylogAuditLogClient
+from snuba.clickhouse.native import ClickhouseResult
+from snuba.clusters.cluster import ClickhouseClientSettings
+from snuba.datasets.schemas.tables import TableSchema
+from snuba.datasets.storages.factory import get_storage
+from snuba.datasets.storages.storage_key import StorageKey
+from snuba.state import get_config
+
+ENABLE_QUERYLOG_API_CONFIG = "enable_clickhouse_querylog_api"
+
+querylog_audit_log_notification_client = QuerylogAuditLogClient()
+
+
+def run_querylog_query(query: str, user: str) -> ClickhouseResult:
+    """
+    Validates, audit logs, and executes given query against Querylog
+    table in ClickHouse.
+    """
+    schema = get_storage(StorageKey.QUERYLOG).get_schema()
+    assert isinstance(schema, TableSchema)
+    validate_ro_query(sql_query=query, allowed_tables=[schema.get_table_name()])
+    if get_config(ENABLE_QUERYLOG_API_CONFIG, 0):
+        audit_log_query(query, user)
```

Comment:
Can we instead make a decorator like 

```python3
@auditlog_func
def run_querylog_query(...):
    pass
```

This decorator should log every call to said function and whether it errored. That way we can be sure nobody forgets to put it in the right place if the function changes

Reply:
You can do it in this PR or file a ticket to do it later